elegant/gui/general_pose_annotator.py

```diff
@@ -0,0 +1,95 @@
+import freeimage
+import pathlib
+import pickle
+import pkg_resources
+from PyQt5 import Qt
+import numpy
+from functools import partial
+
+from ris_widget import ris_widget
+from ris_widget import shared_resources
+
+from elegant.gui import pose_annotation
+from elegant import worm_widths
+from elegant import worm_spline
+
+from zplib.image import colorize
+
+def save_annotations(rw):
+    """Save the pose annotations as pickle files into the parent directory.
+    A pickle file is created for each page in the flipbook with the name of the first image in the 
+    flipbook_page list as the base for the pickle file name.
+    """
+    for fp in rw.flipbook_pages:
+        annotations = fp.annotations
```

Comment:
the above will give AttributeError if the flipbook page hasn't been visited yet by the annotator (which adds the `.annotations` attribute on the fly). I suggest using `getattr(fp, 'annotations', None)` and if the value returned is None, just don't save anything out. You might also want to check that any returned annotation dict (a) has a key for `pose` and (b) that the value is not `None` or `(None, None)`. In either of these conditions just don't save anything either.